src/cmd/flux-kvs.c

```diff
@@ -1054,6 +1054,27 @@ int cmd_dropcache (optparse_t *p, int argc, char **argv)
     return (0);
 }
 
+static char *process_key (const char *key, char **key_suffix)
+{
+    char *nkey;
+    char *ptr;
+
+    if (!(nkey = malloc (strlen (key) + 2))) // room for decoration char + null
+        log_err_exit ("malloc");
+    strcpy (nkey, key);
+
+    if (!strncmp (nkey, "ns:", 3)
+        && (ptr = strchr (nkey, '/'))) {
+        /* No key suffix, decorate with default '.' */
+        if (*(ptr + 1) == '\0')
+            strcat (nkey, ".");
```

Comment:
Because you decorate with `.` here, is that resulting in an extra `.` when the keys are actually printed in `dir`? e.g. I'm seeing:

```
$ flux kvs put ns:foo/bar=none
$ flux kvs dir ns:foo/
ns:foo/..bar = none
```

The `/..` might confusing people. I would assume we'd want the same output as we used to put the key in the first place, i.e. `ns:foo/bar = none` -- leading dots stripped.

Reply:
As long as you are in here, I wonder if a fix for this issue would also fix #1391? For some reason I find this awkward:
```
$ flux kvs dir resource. 
resource..hwloc.
```

Possibly just stripping all trailing `.` from "key" will help?

Reply:
Oops sorry, @garlick beat me to it.

Reply:
Yup, that is issue #1391.  I was going to fix it separately.